### DISCUSSION TITLE
Thermodynamic Database Enthalpy Calculation

### DISCUSSION THREAD

**Discussion by aaronmlarsen** [mutationpp/Mutationpp#264] [Q&A] [answered]
Within the RRHO thermo database, when calculating enthalpy of translation and rotation, the value is multiplied by temperature. However, when this is called through mppequil or the different state models, it calls **speciesHOverRT** and then multiplies by Ru and T, effectively making these enthalpy values T^2 rather than just T. Is there a reason the thermo database is written like this?

**Comment by jbscoggi** [comment, marked as answer]
Hi @aaronmlarsen, thank you for your question.  If you have a look at https://github.com/mutationpp/Mutationpp/blob/aa017852f4d0d19ef561b6a1c1b2638d6366fb4c/src/thermo/RrhoDB.cpp#L269 you will see that the enthalpy is divided by the heavy particle translational temperature before being returned by the `RrhoDB` object.  Similar divisions are incorporated in the other branches of that function which compute mode-specific enthalpies.  We do it this way because in a multi-temperature setting, the temperature used to compute each different energy mode may be different but we always non-dimensionalize by the heavy particle translational temperature.  To generalize to any set of temperatures, we just compute the enthalpies (in units of K) and then divide by `Th` at the end, before returning.  The RRHO database has been validated against other thermodynamic databases and has a number of unit tests which ensure the behavior does not inadvertently change in future pull requests.  I hope that answers your question.
